packages/python/plotly/plotly/tests/test_optional/test_px/test_facets.py

```diff
@@ -1,12 +1,13 @@
-import pandas as pd
 import plotly.express as px
 from pytest import approx
 import pytest
 import random
 
 
-def test_facets():
-    df = px.data.tips()
+def test_facets(constructor):
+    data = px.data.tips().to_dict(orient="list")
```

Comment:
What if the constructor functions were modified to accept either a Pandas df or a dict? Then if a Pandas df is passed it could call `.to_dict()` on it. Would save a bit of boilerplate in the tests.

Reply:
Yes it makes sense. However let me mention that after our meeting today I started working to allow data(set) module to be dataframe agnostic. Different PR should be ready tomorrow and it may be a quick win to simplify data loading in tests as well. I will ping you there